src/main/java/fi/digitraffic/tis/vaco/validation/RulesetSubmissionService.java

```diff
@@ -124,38 +124,76 @@ void submitTask(Entry entry,
         }
 
         String identifyingName = r.identifyingName();
+
+        // mark the processing of matching task as started
+        Optional<Task> ruleTask = taskService.findTask(entry.publicId(), identifyingName);
+        ruleTask.map(t -> taskService.trackTask(entry, t, ProcessingState.START))
+            .orElseThrow();
```

Comment:
With some night's sleep on this, I think this snippet should be at the start of `submit()` method, not here. This was moved out of the scope of that method yesterday while other fixes were done and upon further thought I think it should be moved back to where it came from.